base/array.jl

```diff
@@ -827,6 +836,18 @@ function deleteat!(a::Vector, inds)
     return a
 end
 
+function deleteat!(a::Vector, inds::AbstractVector{Bool})
```

Comment:
I'm not sure what all is defined for `deleteat!`, but it may be possible to make this signature way more general and simpler with `to_indices`.  It could be:

```julia
_deleteat!(a::Vector, I::Integer) = # int method
_deleteat!(a::Vector, I::AbstractArray) = # existing method above
deleteat!(a::Vector, inds) = _deleteat!(a, to_indices(a, (inds,))[1]) # to_indices must return either Integers or AbstractArray of integers
```